Adds Optional.publisher as a convenience for Optional.Publisher.init. (#32)
CombineCommunity/CombineExt: README.md

5
```

## Convenience

## Optional.publisher

> `Optional.publisher` is a method version of [`Optional.Publisher.init`](https://developer.apple.com/documentation/swift/optional/publisher/3343960-init). It puts the type on equal footing with [`Result.publisher`](https://developer.apple.com/documentation/swift/result/3344716-publisher).

Reviewer: ```suggestion
`Optional.publisher` is a property version of [`Optional.Publisher.init`](https://developer.apple.com/documentation/swift/optional/publisher/3343960-init). It puts the type on equal footing with [`Result.publisher`](https://developer.apple.com/documentation/swift/result/3344716-publisher) and [`Sequence.publisher`](link here...).
```

> ```swift
> Optional(1)
>   .publisher
>   /* … */
> ```
> 
> instead of
> 
> ```swift
> Optional.Publisher(Optional(1))
>   .publisher
>   /* … */
> ```

Reviewer: Maybe like this? 

```suggestion
let number: Int? = 1
number.publisher
  /* … */

instead of

let number: Int? = 1
Optional.Publisher(number)
  /* … */
```

> instead of

Reviewer: ```suggestion
Instead of:
```
